pkg/metrics/metrics.go

```diff
@@ -351,20 +405,66 @@ func (c *Counter) IsAlertFiring(alert string, checkedNS, ignoredNS []string) (bo
 	return false, nil
 }
 
-func getPromHost(c client.Client) (*string, error) {
+// GetServiceEndpoint accepts a client,namespace,svcName and portName and attempts to retrive
+// the services endpoint in the form of resolveable.service:portnumber.
+func GetServiceEndpoint(c client.Client, namespace, svcName, portName string) (string, error) {
+	svc := &corev1.Service{}
+	err := c.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: svcName}, svc)
+	if err != nil {
+		return "", err
+	}
+
+	host := fmt.Sprintf(svcName + "." + namespace + clusterSVCSuffix)
+	var port string
+	for _, p := range svc.Spec.Ports {
+		if p.Name == portName {
+			port = strconv.FormatInt(int64(p.Port), 10)
+		}
+	}
+	endpoint := fmt.Sprint(host + ":" + port)
+	return endpoint, nil
+}
+
+func isRunModeLocal() bool {
+	return os.Getenv(k8sutil.ForceRunModeEnv) == string(k8sutil.LocalRunMode)
+}
+
+func Endpoint(c client.Client, namespace, appName, portName string) (string, error) {
+	var endpoint string
+	var err error
+
+	runLocal := isRunModeLocal()
+	useRoutes := config.UseRoutes()
+
+	if !runLocal || runLocal && !useRoutes {
+		endpoint, err = GetServiceEndpoint(c, namespace, appName, portName)
+		if err != nil {
+			return endpoint, err
+		}
+	} else {
+		endpoint, err = getRouteEndpoint(c, appName)
+		if err != nil {
+			return endpoint, err
+		}
+	}
+
+	return endpoint, nil
+}
+
+func getRouteEndpoint(c client.Client, appName string) (string, error) {
```

Comment:
function name to `getRouteHostName` might be less confusing

Reply:
updated.